src/sas/sascalc/file_converter/bsl_loader.py

```diff
@@ -76,8 +76,67 @@ def __init__(self, filename):
         if not is_valid:
             raise BSLParsingError(err_msg)
 
-        CLoader.__init__(self, data_info['filename'], data_info['frames'],
-            data_info['pixels'], data_info['rasters'], data_info['swap_bytes'])
+        self.filename = filename
+        self.n_frames = frames
+        self.n_pixels = pixels
+        self.n_rasters = rasters
+        self.swap_bytes = swap_bytes
```

Comment:
Changed, and in the original implementation filename is set to data_info['filename'], so what exactly do you mean by filename does not match the 'filename' entry?

Reply:
I meant that` filename` happened to be defined in this function (unlike frames, pixels, rasters and swap_bytes).  This might have caused confusion if you decided to make these all local variables rather than keeping them in the data_info dictionary.

Reply:
I see, should be fine, I'll test with the 4.x gui to be sure.